lightningd/gossip/gossip.c

```diff
@@ -198,6 +204,14 @@ static void queue_pkt(struct peer *peer, u8 *msg)
  * queued gossip messages and processes them. */
 static struct io_plan *pkt_out(struct io_conn *conn, struct peer *peer);
 
+/**
+ * peer_dump_gossip_nonlocal - Regularly forward broadcasts to a nonlocal peer
+ * If we are non-local then we just pipe all messages
+ * to the main-daemon synchronously, no need to use
+ * non-blocking IO
```

Comment:
Or we have a dedicated gossip fd to each peer, and avoid the bouncing altogether?